src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1389,6 +1389,37 @@ EnsureFunctionOwner(Oid functionId)
 }
 
 
+/*
+ * EnsureDistributedTable error out if the given relation is not a distributed table
+ * with the given message.
+ */
+void
+EnsureDistributedTable(Oid sourceRelationId, const char *errorMessage)
+{
+	if (!IsDistributedTable(sourceRelationId))
```

Comment:
what about append-distributed tables?

Reply:
Only hash distributed tables can be colocated, so I will update them.